reference-adapter/pspcl-ifix-adapter/src/main/java/org/egov/mapper/PspclDataEntityMapper.java

```diff
@@ -0,0 +1,268 @@
+package org.egov.mapper;
+
+import client.stub.GetBillResult;
+import client.stub.GetPaymentResult;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.entity.BillJsonData;
+import org.egov.entity.PaymentJsonData;
+import org.egov.entity.PspclBillDetail;
+import org.egov.entity.PspclPaymentDetail;
+import org.egov.util.PspclIfixAdapterUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import static org.egov.util.PspclIfixAdapterConstant.BILL_ISSUE_DATE_FORMAT;
+import static org.egov.util.PspclIfixAdapterConstant.TXN_DATE_FORMAT;
+
+@Component
+@Slf4j
+@AllArgsConstructor
+@NoArgsConstructor
+public class PspclDataEntityMapper {
+
+    @Autowired
+    private PspclIfixAdapterUtil pspclIfixAdapterUtil;
+
+
+    public PspclBillDetail mapPspclBillToEntity(GetBillResult currentMonthBill) {
+        PspclBillDetail pspclBillDetail = new PspclBillDetail();
+        BillJsonData billJsonData = new BillJsonData();
+
+        billJsonData.setBILL_CYCLE(currentMonthBill.getBILL_CYCLE());
```

Comment:
For one-to-one mapping, isn't there any copy function provided by spring? There are high chances of missing out because of human error when we are copying 100 attributes. Try to search if it is possible.

Reply:
done.